types.go

```diff
@@ -24,6 +24,108 @@ func (t Timestamp) Parse() (time.Time, error) {
 	return time.Parse(time.RFC3339, string(t))
 }
 
+// NullString is a custom String type to allow for optional modifying strings that can be reset
+// the JSON Marshalling encoding is unique:
+// if the string is empty, it will be encoded as null
+// if the string is not empty, it will be encoded as a string
+type NullString string
+
+// MarshalJSON returns the JSON encoding of the NullString
+func (s NullString) MarshalJSON() ([]byte, error) {
+	if s == "" {
+		return json.Marshal(nil)
+	}
+
+	return json.Marshal(string(s))
+}
+
+func nullString(v interface{}) *NullString {
+	switch v.(type) {
+	case string:
+		if v.(string) == "" {
+			return nil
+		}
+
+		ns := NullString(v.(string))
+		return &ns
+	case NullString:
+		ns := NullString(v.(NullString))
+		return &ns
+	case *NullString:
+		return v.(*NullString)
+	default:
+		return nil
+	}
+}
+
+// NullBool is a custom Bool type to allow for optional modifying booleans that can have false as value
+// the JSON Marshalling encoding is unique:
+// if the bool is false, it will be encoded as false
+// if the bool is true, it will be encoded as true
+type NullBool bool
+
+// MarshalJSON returns the JSON encoding of the NullBool
+func (b NullBool) MarshalJSON() ([]byte, error) {
+	if b == false {
+		return json.Marshal(false)
+	}
+
+	return json.Marshal(true)
+}
```

Comment:
Can this be simplified to the following?

```suggestion
func (b NullBool) MarshalJSON() ([]byte, error) {
	return json.Marshal(b)
}
```

Or should this be the following where if `b` is false, we marshal `nil`? If not, I'm not sure I understand how this would operate differently from a regular `bool`.

```suggestion
func (b NullBool) MarshalJSON() ([]byte, error) {
	if b {
		return json.Marshal(b)
	}

	return json.Marshal(nil)
}
```